custom_components/myskoda/coordinator.py

```diff
@@ -168,38 +169,35 @@ async def _on_operation_event(self, event: EventOperation) -> None:
             await self.update_status(immediate=True)
 
     async def _on_charging_event(self, event: EventCharging):
-        # TODO @webspider: Refactor with proper classes
         vehicle = self.data.vehicle
-
         data = event.event.data
 
         if vehicle.charging is None or vehicle.charging.status is None:
             await self.update_charging()
         else:
-            status = vehicle.charging.status
-
-            if data.charged_range is not None:
-                status.battery.remaining_cruising_range_in_meters = (
-                    data.charged_range * 1000
-                )
-            if data.soc is not None:
-                status.battery.state_of_charge_in_percent = data.soc
-            if data.time_to_finish is not None:
-                status.remaining_time_to_fully_charged_in_minutes = data.time_to_finish
-            if data.state is not None:
-                status.state = data.state
+            # TODO: support other charging events
+            match event.event:
+                case ServiceEventChangeSoc():
+                    status = vehicle.charging.status
+                    status.battery.remaining_cruising_range_in_meters = (
+                        data.charged_range * 1000
+                    )
+                    status.battery.state_of_charge_in_percent = data.soc
+                    if data.time_to_finish is not None:
+                        status.remaining_time_to_fully_charged_in_minutes = (
+                            data.time_to_finish
+                        )
+                        status.state = data.state
```

Comment:
Do we want to set `last_updated` to `timestamp` here?

Reply:
Sorry, could you clarify? I could not find the `last_updated` field anywhere in the repo.

Reply:
I meant the LastUpdated(MySkodaSensor)